### PR TITLE
PHP: Remove WP <5.5 compatibility code 

### LINKED ISSUE #7600: PHP: Remove WP <5.5 compatibility code
## Task Description

Places that come to mind:

#7531 

https://github.com/google/web-stories-wp/blob/c4b2f02b98713b17962f283a5f2205f93d56fbce/includes/Admin/Cross_Origin_Isolation.php#L276-L282

There might be more

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

PHP: Remove WP <5.5 compatibility code 

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #7600

### CODE REVIEW

File: includes/KSES.php [hunk 1]
```diff
@@ -54,20 +54,7 @@ public function register() {
...
-		// The third param of the below filter was only added in WP 5.4.1.
-		if ( version_compare( get_bloginfo( 'version' ), '5.4.1', '>=' ) ) {
-			add_filter( 'wp_insert_post_data', [ $this, 'filter_insert_post_data' ], 10, 3 );
-
-			return;
-		}
-
-		// Fallback for WP < 5.4.1. Affects all post types, not just the web-story one.
-		// TODO: Remove once support for WP 5.3 is dropped.
-		add_filter( 'safe_style_css', [ $this, 'filter_safe_style_css' ] );
```
Comment (miina) [inline includes/KSES.php:66]: Should any of these methods be fully removed, too? Or are these still used elsewhere? I'm seeing other usages of some of these filters but are these still necessary?

Also, seeing this comment below in this file for `safecss_filter_attr`:
`Use safe_style_disallowed_chars filter once WP 5.5+ is required.`

Should this method (`safecss_filter_attr`) be replaced with `safe_style_disallowed_chars`? Or is this done in another PR?

Comment (swissspidy) [reply includes/KSES.php:66]: +1 to removing unused methods.

As for safe_style_disallowed_chars, it should be done in a separate ticket, but I don‘t think it actually makes a lot of sense maintenance wise

Comment (spacedmonkey) [reply includes/KSES.php:66]: `filter_safe_style_css` is used in `filter_insert_post_data`. Why would be remove it?

> As for safe_style_disallowed_chars, it should be done in a separate ticket, but I don‘t think it actually makes a lot of sense maintenance wise

Agree, this is for another ticket.

File: includes/Admin/Cross_Origin_Isolation.php [hunk 2]
```diff
@@ -310,12 +310,6 @@ protected function is_needed() {
...
 		}
 
-		// Backwards compatible, default values added in WP 5.5.
-		$check = get_user_meta( $user_id, Preferences::MEDIA_OPTIMIZATION_META_KEY, false );
-		if ( empty( $check ) ) {
-			return true;
-		}
-
-		return rest_sanitize_boolean( $check[0] );
+		return get_user_meta( $user_id, Preferences::MEDIA_OPTIMIZATION_META_KEY, true );
```
Comment (miina) [inline includes/Admin/Cross_Origin_Isolation.php]: Should this be `(bool) get_user_meta(...)`?

File: includes/KSES.php (hunk 1 above)
Comment (miina) [reply includes/KSES.php:66]: > Agree, this is for another ticket.

Should we create one? Or remove the comment if it won't be necessary?

File: includes/Admin/Cross_Origin_Isolation.php (hunk 2 above)
Comment (spacedmonkey) [reply includes/Admin/Cross_Origin_Isolation.php]: I added back the `rest_sanitize_boolean` function call.

File: includes/KSES.php (hunk 1 above)
Comment (spacedmonkey) [reply includes/KSES.php:66]: Follow up issue. https://github.com/google/web-stories-wp/issues/7992

**Review by miina — APPROVED** [review]

**Comment by miina** [comment]
This was blocked by https://github.com/google/web-stories-wp/pull/7926, no? :)

**Pull request merged**